### PR TITLE
test(builder): add standalone build_block unit test with in-memory state

### PR DESCRIPTION
**Pull request by mw2000**
## Summary

- Add `OpPayloadBuilderCtx::for_test(chain_spec, parent)` constructor behind `#[cfg(any(test, feature = "test-utils"))]` so the builder hot path can be tested without booting a full reth node
- Widen `build_block` and `execute_pre_steps` from `pub(super)` / private to `pub(crate)` so tests elsewhere in the crate can call them directly
- Expose `test_utils::instance::chain_spec()` as `pub` for reuse by test constructors
- Add `build_block_empty_no_state_root` test that verifies empty-block assembly (header, receipts root, flashblocks payload) using `State<StateProviderDatabase<NoopProvider>>` — no disk, no network, no `LocalInstance`

## Test plan

```
cargo test -p base-builder-core --lib
# 44 passed, 0 failed
cargo clippy -p base-builder-core --tests
# 0 warnings
```

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>✅ Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
✅
<code>1/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td><code>2</code> if repo is sensitive</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>1</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>2</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: crates/builder/core/src/test_utils/instance.rs [hunk 1]
```diff
@@ -296,7 +296,9 @@ pub fn default_node_config() -> NodeConfig<OpChainSpec> {
     node_config_with_chain_spec(chain_spec())
 }
 
-fn chain_spec() -> Arc<OpChainSpec> {
+/// Returns the default test chain spec, lazily initialized from the embedded
+/// genesis template.
+pub fn chain_spec() -> Arc<OpChainSpec> {
```
[the remark this answers is not in the document]
Comment (mw2000) [reply crates/builder/core/src/test_utils/instance.rs:301]: Will be using this in follow ups

File: crates/builder/core/src/flashblocks/payload.rs [hunk 2]
```diff
@@ -899,7 +899,7 @@ struct FlashblocksMetadata {
     access_list: Option<FlashblockAccessList>,
 }
 
-fn execute_pre_steps<DB>(
+pub(crate) fn execute_pre_steps<DB>(
```
[the remark this answers is not in the document]
Comment (mw2000) [reply crates/builder/core/src/flashblocks/payload.rs:902]: Will be using this in follow ups

**Review by refcell — APPROVED** [review]
Nice!
